src/repository/note.repository.ts

```diff
@@ -81,4 +81,13 @@ export default class NoteRepository {
   public async getNoteListByUserId(id: number, offset: number, limit: number): Promise<Note[]> {
     return await this.storage.getNoteListByUserId(id, offset, limit);
   }
+
+  /**
+   * Get all note parents based on note id
+   * @param noteId : note id to get all its parents
+   * @returns an array of note parents objects containing public id and content
+   */
+  public async getNoteParents(noteId: NoteInternalId): Promise<Note[]> {
```

Comment:
1. I'd suggest to move this method to the `NoteRelationsRepository` since it is responsible for working with note relations. 

2. And remove added association from Note Storage.

3. Repository can return a list of `NoteInternalId`

4. Then on Domain level (in a Note Service) you can access Note Repository for getting Notes by ids. 

5. For doing this, you can implement getNotesByIds method in Note Repository and Store.